Calculation of Design strength of Lag Screws using CSA O86-09

In [1]:
import pint
unit = pint.UnitRegistry()

# define synanyms for common units
inch = unit.inch; mm = unit.mm; m = unit.m; kPa = unit.kPa; MPa = unit.MPa; 
psi = unit.psi; kN = unit.kN; ksi = unit.ksi; 
dimensionless = unit.dimensionless; s = unit.second; kg = unit.kg

def u_round(Q, digits=3):
    """
    Takes a Pint.py quantity and returns same rounded to digits, 
    default is 3 digits
    """
    try:
        unit.check(Q.units)
        magnitude = Q.magnitude
        units = Q.units
        return round(magnitude,digits) * units
    except:
        print('ERROR: u_round() first arguement must be a Pint.py quantity')
        return NaN

In [2]:
# Define case parameters
G = 0.50 # mean density for Douglas-Fir-Larch from Table A10.1
t_1 = 38*mm # side member thickness
t_2 = 76*mm # main member thickness
d_F = (3/8) * inch # lag screw shank diameter
lag_L = 100*mm
d_F.ito(mm)
lag_E = 6.4*mm # length of tip, from Wood Design Manual 2010 Table 7.15
L_p = lag_L - (t_1 + lag_E)
K_D = 1.15 # Duration factor, short-term, less than 7 days
K_SF = 1.0 # Dry service conditions
K_T = 1.0 # untreated wood
J_E = 1.0 # screws installed in side grain; not end grain
Kprime = K_D * K_SF * K_T

print('Case parameters:')
print('Mean density of Douglas-Fir-Larch from Table A10.1, G =',round(G,2))
print('Thickness of side member, t_1 =',u_round(t_1,1))
print('Thickness of main member, t_2 =',u_round(t_2,1))
print('Lag Screw diameter, d_F =',u_round(d_F,1))
print('Lag Screw length, L =',u_round(lag_L,1))
print('Lag Screw tip length, E =',u_round(lag_E,1))
print('Lag Screw Penetration, L_p = L - (t_1 + E) =',u_round(L_p,1))
print('')
print('Modification Factors')
print('Duration Factor, K_D =',round(K_D,2))
print('Service Condition Factor, K_SF =',round(K_SF,1))
print('Fire-retardant Treatment Factor, K_T =',round(K_T,1))
print('Grain Factor, J_E =',round(J_E,1))
print('K` = K_D * K_SF * K_T =',round(Kprime,2))

Case parameters:
Mean density of Douglas-Fir-Larch from Table A10.1, G = 0.5
Thickness of side member, t_1 = 38 millimeter
Thickness of main member, t_2 = 76 millimeter
Lag Screw diameter, d_F = 9.5 millimeter
Lag Screw length, L = 100 millimeter
Lag Screw tip length, E = 6.4 millimeter
Lag Screw Penetration, L_p = L - (t_1 + E) = 55.6 millimeter

Modification Factors
Duration Factor, K_D = 1.15
Service Condition Factor, K_SF = 1.0
Fire-retardant Treatment Factor, K_T = 1.0
Grain Factor, J_E = 1.0
K` = K_D * K_SF * K_T = 1.15


In [3]:
Qprime_r = 1.08*kN
Q_r = Qprime_r * Kprime
print('Per WDM2010 Leg Screw Selection Table, Q`_r = ',u_round(Qprime_r,2))
print('Q_r = Q`_r * K`')
print('Q_r =',u_round(Q_r,2),'per lag screw')

Per WDM2010 Leg Screw Selection Table, Q`_r =  1.08 kilonewton
Q_r = Q`_r * K`
Q_r = 1.24 kilonewton per lag screw


**Given:** 2 kPa Live Load  
**Given:** Max 1.3m width of platform  
**Given:** Lag Screw capacity above  
**Given:** Safety Factor of 1.5:1  
**Find:** Max on-centre spacing of lag screws

$1.24 kN = (2 kPa)(1.5)(1.3 m / 2)(x)$  
$x = 1.24 kN / ((2 kPa)(1.5)(0.65 m))$  

In [4]:
x = u_round((Q_r / ((2*kPa)*(1.5)*(0.65*m))).to(m),3)
print('Given conditions above and 3/8" x 4" Lag screws',
      'screws should be a max',x,'on-centre.')

Given conditions above and 3/8" x 4" Lag screws screws should be a max 0.637 meter on-centre.


**Given:** Conditions stated above  
**Find:** Max on-centre spacing when 3/8" x 3" lag screws are used

In [5]:
lag_L = 76*mm
lag_E = 6.4*mm
L_p = lag_L - (t_1 + lag_E)
print('Lag Screw Penetration, L_p = L - (t_1 + E) =',u_round(L_p,1))

Lag Screw Penetration, L_p = L - (t_1 + E) = 31.6 millimeter


In [6]:
# O86-09 10.6.3.3  
"""
For lag screws loaded laterally, the minimum length of penetration into the
main member (t_2 in Clause 10.6.6.1.2) shall be not less than five times the
shank diameter, d_F
"""
print('5 x shank diameter =',u_round(5*d_F,1))

5 x shank diameter = 47.6 millimeter


Leg Screw Penetration is less than minimum required penetration. Therefore, 3/8" x 3" Lag Screws are NFG